### DISCUSSION TITLE
What would happen if a span in opentelmetry-java is never be ended?

### DISCUSSION THREAD

**Discussion by aaron-ai** [open-telemetry/opentelemetry-java#3288] [Q&A]
would cause memory leak in Java? my application has asynchronous call, which may not ensure that the span is ended every time.

**Comment by Oberon00** [comment]
It depends (mainly) on the SpanProcessors used. The default ones only do something when the span is ended, so starting a span and not ending it might not cause any issues in practice. But the SDK/agent is allowed by the spec  to have memory leaks if the span is not ended. So it is highly advisable to find a way to always end the span.
